wp-content/themes/core/components/blocks/gallery_grid/Gallery_Grid_Controller.php

```diff
@@ -0,0 +1,346 @@
+		];
+	}
+
+	public function get_block_id(): string {
+		return $this->id;
+	}
+
+	public function get_classes(): string {
+		return Markup_Utils::class_attribute( $this->classes );
+	}
+
+	public function get_attributes(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function is_slideshow(): bool {
+		return $this->slideshow;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		$this->content_classes[] = 'gallery-layout--' . $this->grid_layout;
+	
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
+			return [];
+		}
+
+		return [
+			Content_Block_Controller::TAG     => 'header',
+			Content_Block_Controller::TITLE   => $this->get_title(),
+			Content_Block_Controller::CONTENT => $this->get_content(),
+			Content_Block_Controller::CLASSES => [
+				'c-block__content-block',
+				'c-block__header',
+				'b-gallery-grid'
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_title(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::TAG     => 'h2',
+			Text_Controller::CLASSES => [
+				'c-block__title',
+				'b-gallery-grid__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_slideshow_title() {
+		return $this->title ?? '';
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-gallery-grid__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_slideshow_button() {
+		return [
+			Button_Controller::CLASSES    => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after',
+				'icon-arrow-right'
+			],
+			Button_Controller::ATTRS      => [ 'data-js'  => 'dialog-trigger', 'data-content' => 'dialog-content-' . $this->id ],
+			Button_Controller::CONTENT    => esc_html__( 'View slideshow', 'tribe' ),
+		];
+	}
+
+	/**
+	 * Get the Slider
+	 *
+	 * @return array
+	 */
+	public function get_slider_args(): array {
+		$main_attrs = [];
+
+		$swiper_options = '{"preloadImages":"false","lazy":"true","watchSlidesVisibility":"true","spaceBetween":60,"ally":"true","keyboard":"true","grabCursor":"true","navigation":{"nextEl":".swiper-button-next","prevEl":".swiper-button-prev"}}';
```

Comment:
Can you adjust this to use a PHP array to define the options, then use `json_encode()` to encode them into the JSON string before sending it along? It makes reading or editing the options / config much easier later.  [Here's an example from Conroe](https://github.com/ConroeISD/external-website/blob/sprint/2020.11/wp-content/plugins/core/src/Templates/Panels/MediaGallery.php#L232-L244) (uses a different slider lib, but the concept is the same.)